build_tools/aws-sdk-code-generator/spec/protocols/input/json.json

```diff
@@ -491,6 +491,145 @@
       }
     ]
   },
+  {
+    "description": "Document Type",
```

Comment:
Are these ported from Smithy?

Reply:
They are loosely based on the smithy protocol tests.